src/login/logind.c

```diff
@@ -455,8 +455,6 @@ static int deliver_session_leader_fd_consume(Session *s, const char *fdname, int
                 return r;
         }
 
-        assert(pid_is_valid(s->deserialized_pid));
```

Comment:
With the check added in the caller, this can be kept now, no?

Reply:
I rearranged the code instead. PTAL.